xarray/tests/test_concat.py

```diff
@@ -558,3 +558,36 @@ def test_concat_merge_single_non_dim_coord():
     for coords in ["different", "all"]:
         with raises_regex(ValueError, "'y' not present in all datasets"):
             concat([da1, da2, da3], dim="x")
+
+
+def test_concat_preserve_coordinate_order():
+    x = np.arange(0, 5)
+    y = np.arange(0, 10)
+    time = [0, 1]
+    data = np.zeros((2, 10, 5), dtype=bool)
+
+    ds1 = Dataset(
+        {"data": (["time", "y", "x"], data[0])},
+        coords={"time": time[0], "y": y, "x": x},
+    )
+    ds2 = Dataset(
+        {"data": (["time", "y", "x"], data[1])},
+        coords={"time": time[1], "y": y, "x": x},
+    )
```

Comment:
turns out I removed too much:
```suggestion
    ds1 = Dataset(
        {"data": (["time", "y", "x"], [data[0]])},
        coords={"time": time[0], "y": y, "x": x},
    )
    ds2 = Dataset(
        {"data": (["time", "y", "x"], [data[1]])},
        coords={"time": time[1], "y": y, "x": x},
    )
```